src/main/java/org/opentripplanner/middleware/triptracker/NotifyBusOperator.java

```diff
@@ -0,0 +1,196 @@
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.opentripplanner.middleware.triptracker.TripStatus.getSegmentStartTime;
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+
+/**
+ * If conditions are correct notify a bus operator of a traveler joining the service at a given stop.
+ */
+public class NotifyBusOperator {
+
+    public NotifyBusOperator() {}
+
+    public static boolean IS_TEST = false;
+
+    private static final Logger LOG = LoggerFactory.getLogger(NotifyBusOperator.class);
+
+    private static final String BUS_OPERATOR_NOTIFIER_API_URL
+        = getConfigPropertyAsText("BUS_OPERATOR_NOTIFIER_API_URL", "not-provided");
+
+    private static final String BUS_OPERATOR_NOTIFIER_API_SUBSCRIPTION_KEY
+        = getConfigPropertyAsText("BUS_OPERATOR_NOTIFIER_API_SUBSCRIPTION_KEY", "not-provided");
+
+    public static List<String> QUALIFYING_BUS_NOTIFIER_ROUTES = getBusOperatorNotifierQualifyingRoutes();
+
+    public static final int ACCEPTABLE_AHEAD_OF_SCHEDULE_IN_MINUTES = 15;
+
+    /**
+     * Headers that are required for each request.
+     */
+    private static final Map<String, String> BUS_OPERATOR_NOTIFIER_API_HEADERS = Map.of(
+        "Ocp-Apim-Subscription-Key", BUS_OPERATOR_NOTIFIER_API_SUBSCRIPTION_KEY,
+        "Content-Type", "application/json"
+    );
+
+
+    /**
+     * Get the routes which qualify for bus operator notifying from the configuration. The configuration value is
+     * expected to be a comma separate list of agency id (as provided by OTP not agency) and route id e.g.
+     * <p>
+     * GwinnettCountyTransit:360,GwinnettCountyTransit:40,GwinnettCountyTransit:25
+     */
+    private static List<String> getBusOperatorNotifierQualifyingRoutes() {
+        String busOperatorNotifierQualifyingRoutes
+            = getConfigPropertyAsText("BUS_OPERATOR_NOTIFIER_QUALIFYING_ROUTES");
+        if (busOperatorNotifierQualifyingRoutes != null) {
+            return Arrays.asList(busOperatorNotifierQualifyingRoutes.split(","));
+        }
+        return new ArrayList<>();
+    }
+
+    /**
+     * Stage notification to bus operator by making sure all required conditions are met.
+     */
+    public static void sendNotification(TripStatus tripStatus, TravelerPosition travelerPosition) {
+        try {
+            if (
+                isBusLeg(travelerPosition.nextLeg) &&
+                isWithinOperationalNotifyWindow(tripStatus, travelerPosition) &&
+                hasNotPreviouslyNotifiedBusDriverForRoute(travelerPosition.trackedJourney, travelerPosition.nextLeg.route.id) &&
+                supportsBusOperatorNotification(travelerPosition.nextLeg.route.id)
+            ) {
+                var body = createPostBody(travelerPosition);
+                var httpStatus = doPost(body);
```

Comment:
We'll have to add a "call pending" state if the call does not complete right away, so that updates that hit while this call is pending don't initiate another one.